src/docset.rs

```diff
@@ -87,6 +106,23 @@ pub trait DocSet: Send {
     /// length of the docset.
     fn size_hint(&self) -> u32;
 
+    /// Returns a best-effort hint of the
+    /// cost to drive the docset.
+    ///
+    /// By default this returns `size_hint()`.
+    ///
+    /// DocSets may have vastly different cost depending on their type,
+    /// e.g. an intersection with 10 hits is much cheaper than
+    /// a phrase search with 10 hits, since it needs to load positions.
+    ///
+    /// ### Future Work
+    /// We may want to differentiate `DocSet` costs more more granular, e.g.
+    /// creation_cost, advance_cost, seek_cost on to get a good estimation
+    /// what query types to choose.
```

Comment:
of course this will always be handwavy, but we might want to give one absolute frame of reference as a target.

for instance 1 unit of cost could be the amortized cost of advancing a term scorer by 1.